### ISSUE TITLE
Not working on Sonoma 14.2

### ISSUE THREAD

**Issue by iGerman00** [Pyroh/Fluor#92]
The app installs and works fine, however the primary function doesn't work, I tried from both brew and the dmg package, but it fails to actually switch the mode. @Pyroh it would be amazing if you could take a look at it.

**Comment by NEKETSKI** [comment]
It works great for me on macos sonoma 14.2.1, air 15 with m2

**Comment by Meligy** [comment]
Works on macOS Sonoma 14.3 (23D56) when installed with homebrew and tested with Parallels for Windows.

**Comment by nycjay** [comment]
I have the same problem, running Fluor 2.5.1 (installed via brew) on my Intel MBP running macOS 14.3.1. The Fluor icon up in the top bar works fine, switching as I move through my apps. But, when I am on an app where I have it configured to use the Fn keys (Eclipse, VS Code, etc), the bar does not show the function key, it's just blank (with the volume controls on the right). When I manually hit the fn key on the keyboard, then the function keys light up on the touch bar. Happy to share any debugging logs, if someone can point me at them.

**Comment by cotwitch** [comment]
> The app installs and works fine, however the primary function doesn't work, I tried from both brew and the dmg package, but it fails to actually switch the mode. @Pyroh it would be amazing if you could take a look at it.

Did you allowed Fluor in Accessibility settings?

<img width="717" alt="image" src="https://github.com/Pyroh/Fluor/assets/10708898/179a7bf9-9ffb-4a36-81ef-21324d3560fd">

**Comment by nycjay** [comment]
Yes, I just checked, and it is allowed. I toggled it off/on just to confirm, but I still get the same behavior.

**Comment by gingerbeardman** [comment]
Working well for me since ever

Currently running macOS Sonoma 14.3.1 with Apple extended wired aluminium keyboard (~16 years old)

**Comment by bbrendon** [comment]
I'm on M1. macOS 15.5 and I can't get it working. I got notifications the mode switches, but the mode doesn't actually change.  Argh!!

**Comment by bbrendon** [comment]
Toggling this in preferences seems to have made it work. Strange.

<img width="440" alt="Image" src="https://github.com/user-attachments/assets/fae10994-df36-468a-8c93-8618fdb7ea91" />

**Comment by bbrendon** [comment]
Just kidding. Apparently F-keys are always enabled. I thought it was working because the pop-up said it was but really, it was just locked on F-keys on.

**Comment by gingerbeardman** [comment]
> but the mode doesn't actually change

What do you expect to see and what are you seeing? What app are you testing with?

For the record, I'm running macOS Sonoma 14.7.6 (23H626) and Fluor 2.5.1 (2391) and all is working as expected. 

Here I have OpenSCAD set to see keys as Fn, but my system default is for the keys to be Brightness.

![Image](https://github.com/user-attachments/assets/b6417e1d-ec60-453e-b2d1-75f1e2627b2a)

Maybe it's worth starting a new issue for macOS 15.x with more details about how exactly it's not working. Feel free to @ me there and we can try to trouble shoot.

**Comment by gonzzza007** [comment]
Can confirm the issue. Sonoma 14.5, MacBook Pro M3, testing with VSCode.

Mostly it works, but sometimes - the icons in toolbar switch but actually Function keys won't work.
When i manually open the settings - switch the mode back and forward - it usually fixes the issue, BUT 
sometimes i have to restart Fluor.

**Comment by gingerbeardman** [comment]
@gonzzza007 what build of Fluor?

**Comment by gonzzza007** [comment]
> [@gonzzza007](https://github.com/gonzzza007) what build of Fluor?

I'm really sorry for my comment, i started to dig in - and found out that another app is messing with my shortcuts!
Fluor (latest) is working fine!
